src/widgets/mainwindow.cpp

```diff
@@ -595,6 +596,17 @@ void MainWindow::closeOnQuit()
 
 void MainWindow::setupShortcuts()
 {
+    const auto &coreConfig = ConfigMgr::getInst().getCoreConfig();
+
+    // For cross-platform global shortcuts, the external library QHotkey is used.
+    auto qHotkey = new QHotkey(QKeySequence(coreConfig.getShortcut(CoreConfig::Global_WakeUp)), true, qApp);
```

Comment:
1. Construct a qkeyseqience first
2. Judge if the sequence is valid
3.if invalid, just give up
4. use this instead of qapp